src/cli/remove.rs

```diff
@@ -44,22 +54,43 @@ pub async fn execute(args: Args) -> miette::Result<()> {
     let results = deps
         .iter()
         .map(|dep| {
-            if let Some(p) = &args.platform {
-                project
-                    .manifest
-                    .remove_target_dependency(dep, &spec_type, p)
+            if args.pypi {
+                DependencyRemovalResult::PyPiDeps(project.manifest.remove_pypi_dependency(dep))
             } else {
-                project.manifest.remove_dependency(dep, &spec_type)
+                DependencyRemovalResult::PixiDeps(if let Some(p) = &args.platform {
+                    project
+                        .manifest
+                        .remove_target_dependency(dep, &spec_type, p)
+                } else {
+                    project.manifest.remove_dependency(dep, &spec_type)
+                })
             }
         })
-        .collect::<Vec<_>>();
+        .collect::<Vec<DependencyRemovalResult>>();
 
     project.save()?;
 
     // updating prefix after removing from toml
     let _ = get_up_to_date_prefix(&project, LockFileUsage::Update).await?;
 
-    for (removed, spec) in results.iter().flatten() {
+    for result in results.iter() {
+        let removed = match result {
+            DependencyRemovalResult::PixiDeps(pixi_result) => {
```

Comment:
You can match on Ok here as well. 

```suggestion
            DependencyRemovalResult::PixiDeps(Ok(name)) => {
```

Reply:
done and refactored